src/HTTP/Cookies/CookieJarWrapper.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace ILIAS\HTTP\Cookies;
+
+use Dflydev\FigCookies\SetCookies;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Class CookieJarFacade
+ *
+ * Wrapper class for the FigCookies SetCookies class.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @package ILIAS\HTTP\Cookies
+ * @since   5.2
+ * @version 1.0.0
+ */
+class CookieJarWrapper implements CookieJar {
+
```

Comment:
Same problem as in CookieWrapper. In this case _with_ and _without_ never create a new instance of CookieJarWrapper. We only get a changed source object.